Sources/ApolloCodegenLib/Frontend/JavaScript/src/compiler/index.ts

```diff
@@ -265,7 +275,7 @@ export function compileToIR(
           ? (typeFromAST(schema, typeNode) as GraphQLCompositeType)
           : parentType;
 
-        referencedTypes.add(typeCondition);
+        addReferencedType(typeCondition);        
```

Comment:
Is it intentional that this use of `typeCondition` doesn't need to be wrapped in `getNamedType`? _I'm not 100% sure whether TS would catch this if it's not a `GraphQLNamedType`._

Reply:
`typeFromAST` returns a named type. It replaces calling `getNamedType`. I'm not 100% sure that it's always right either. But it hasn't broken yet. If this breaks for someone we will know soon in the alpha. I don't think we could even wrap this in `getNamedType` because the `typeCondition` is already a composite (named) type. We get do `getNamedType(typeNode)` possibly. But it's redundant when we've already called `typeFromAST`.